### PR TITLE
dup script

### PR DESCRIPTION
**Pull request by maelle**
cf #144 

cc @jonocarroll 

The script skips all lines after "upcoming events" (maybe not specific enough?) and removes the R code link.

### CODE REVIEW

**Comment by maelle** [comment]
Depending on #590 the script/function might end up in another repo.

**Comment by maelle** [comment]
I added a duplicate by hand in draft.md (ironic given all the ones I had mistakenly added today, and then removed) to check the function.

File: scripts/find_duplicates.R
Comment (jonocarroll) [inline scripts/find_duplicates.R:37]: What about also checking the current week for duplicates - I'm always concerned I'm going to add something twice.

```suggestion
  current_dups <- thisweek[duplicated(thisweek)]
  if (length(current_dups)) {
    message("Duplicates found in current week:")
    cat(paste0(current_dups, collapse = "\n"))
    cat("\n")
  }
  
  dups <- thisweek[thisweek %in% old_links]
```

File: scripts/find_duplicates.R
Comment (jonocarroll) [inline scripts/find_duplicates.R]: Could we allow for the possibility that this list might grow?

```suggestion
  acceptable_dups <- c("http://developer.r-project.org/blosxom.cgi/R-devel/NEWS")
  dups <- setdiff(dups, acceptable_dups)
```

File: scripts/find_duplicates.R
Comment (jonocarroll) [inline scripts/find_duplicates.R:41]: If we add searching for dups in the current issue, maybe no "remaining"?

```suggestion
    message("No remaining duplicate links, well done! :-)")
```

I'm not sure if it's better to return the list or just print it to the screen, either works.

Comment (jonocarroll) [reply scripts/find_duplicates.R:41]: If none are found in the current issue, maybe a message for that, too. Otherwise my "remaining" looks out of place.

**Review by jonocarroll — APPROVED** [review]
Looks great and does what it says on the tin. Cheers!

File: scripts/find_duplicates.R
Comment (maelle) [reply scripts/find_duplicates.R:37]: why don't I ever use `setdiff()`?!

**Comment by maelle** [comment]
what do you mean by current week @jonocarroll? The script looks for links in draft.md that also were in one of the issues from the 20 previous weeks.

**Comment by maelle** [comment]
One thing that script doesn't look for is duplicates within draft.md (same item under different headings), should it be added?

**Comment by jonocarroll** [comment]
That's what the `current_dups` block does -- identifies duplicates within the draft.

**Comment by maelle** [comment]
Right! Thanks! Can we merge this and close the issue or should we get a second review?

**Comment by qinwf** [comment]
LGTM.

If there are small changes in the future, we can push it directly or merge it after reviewed by one reviewer.

**Pull request merged**
